src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocSummaryLocationCheck.java

```diff
@@ -0,0 +1,175 @@
+public class JavadocSummaryLocationCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_SUMMARY_FIRST_LINE = "javadoc.summary.first.line";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_JAVADOC_SUMMARY_SECOND_LINE = "javadoc.summary.second.line";
+
+    /**
+     * Specify the policy on placement of the javadoc summary.
+     */
+    private JavadocSummaryLocation location = JavadocSummaryLocation.SECOND_LINE;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        if (isMultilineComment(ast)) {
+            final JavadocSummaryLocation javadocSummaryLocation = getJavadocSummaryLocation(ast);
+            if (javadocSummaryLocation != location) {
+                if (location == JavadocSummaryLocation.FIRST_LINE) {
+                    log(ast.getLineNumber(), MSG_JAVADOC_SUMMARY_FIRST_LINE);
+                }
+                else {
+                    log(ast.getLineNumber(), MSG_JAVADOC_SUMMARY_SECOND_LINE);
+                }
+            }
+        }
+    }
+
+    /**
+     * Setter to specify the policy on placement of the javadoc summary.
+     * @param value string to decode location from
+     * @throws IllegalArgumentException if unable to decode
+     */
+    public void setLocation(String value) {
+        location = JavadocSummaryLocation.valueOf(value.trim().toUpperCase(Locale.ENGLISH));
+    }
+
+    /**
+     * Checks if a DetailNode of type {@code JavadocTokenTypes.JAVADOC} has
+     * more than one line.
+     *
+     * @param node node to check
+     * @return {@code true} for
+     */
+    private static boolean isMultilineComment(DetailNode node) {
+        final DetailNode[] children = node.getChildren();
+        final int firstChildLineNo = children[0].getLineNumber();
+        final int lastChildLineNo = children[children.length - 1].getLineNumber();
+        return firstChildLineNo != lastChildLineNo;
+    }
+
+    /**
+     * Returns the javadoc summary location for the {@code JavadocTokenTypes.JAVADOC} node.
+     *
+     * @param node node to return
+     * @return {@link JavadocSummaryLocation#FIRST_LINE} for scala-style comments;
+     *         {@link JavadocSummaryLocation#SECOND_LINE} for java-style comments.
+     */
+    private static JavadocSummaryLocation getJavadocSummaryLocation(DetailNode node) {
+        final DetailNode firstChild = JavadocUtil.getFirstChild(node);
+        final JavadocSummaryLocation summaryLocation;
+        if (firstChild.getType() == JavadocTokenTypes.NEWLINE) {
```

Comment:
https://pbludov.github.io/issue-6724-java/android-launcher/index.html#A1
https://pbludov.github.io/issue-6724-java/android-launcher/xref/opt/pbludov/contribution/checkstyle-tester/repositories/android-launcher/src/com/android/launcher/Search.java.html#L345
Space after the `/**` is causing what looks to be a false positive.
It create a tree similar to:
````
    |  |  |  |  `--JAVADOC -> JAVADOC [2:3]
    |  |  |  |      |--TEXT ->  [2:3]
    |  |  |  |      |--NEWLINE -> \n [2:4]
````